File: src/hono-server/routes/webhooks.ts
import { Hono } from 'hono';

import { POST as casdoorWebhook } from '@/handlers/api/webhooks/casdoor/route';
import { POST as logtoWebhook } from '@/handlers/api/webhooks/logto/route';
import { POST as memoryExtractionBenchmark } from '@/handlers/api/webhooks/memory-extraction/benchmark-locomo/route';
import { POST as memoryExtraction } from '@/handlers/api/webhooks/memory-extraction/route';
import { POST as personaUpdateWriting } from '@/handlers/api/webhooks/memory-user-memory/persona/update-writing/route';
import { POST as chatTopicCancel } from '@/handlers/api/webhooks/memory-user-memory/pipelines/extract/chat-topic/cancel/route';
import { POST as videoWebhook } from '@/handlers/api/webhooks/video/[provider]/route';

const webhooks = new Hono();

// ============ Auth Provider Webhooks ============ //
webhooks.post('/api/webhooks/casdoor', (c) => casdoorWebhook(c.req.raw));
webhooks.post('/api/webhooks/logto', (c) => logtoWebhook(c.req.raw));

// ============ Memory Extraction Webhooks ============ //
webhooks.post('/api/webhooks/memory-extraction', (c) => memoryExtraction(c.req.raw));
webhooks.post('/api/webhooks/memory-extraction/benchmark-locomo', (c) =>
  memoryExtractionBenchmark(c.req.raw),
);

// ============ Memory User-Memory Webhooks ============ //
webhooks.post('/api/webhooks/memory-user-memory/persona/update-writing', (c) =>
  personaUpdateWriting(c.req.raw),
);
webhooks.post('/api/webhooks/memory-user-memory/pipelines/extract/chat-topic/cancel', (c) =>
  chatTopicCancel(c.req.raw),
);

// ============ Video Provider Webhooks ============ //
webhooks.post('/api/webhooks/video/:provider', (c) => {
  const provider = c.req.param('provider');
  return videoWebhook(c.req.raw, {
    params: Promise.resolve({ provider }),
  });
});

export default webhooks;

File: python-backend/app/routers/webhooks.py
"""Legacy webhook compatibility endpoints owned by the Python backend."""

from __future__ import annotations

from typing import Any

from fastapi import APIRouter, Request, status
from fastapi.responses import JSONResponse

router = APIRouter(prefix="/api/webhooks", tags=["Webhooks"])


async def _json(request: Request) -> dict[str, Any]:
    try:
        body = await request.json()
    except Exception:
        return {}
    return body if isinstance(body, dict) else {}


@router.post("/casdoor")
@router.post("/logto")
async def retired_auth_provider_webhook(request: Request):
    await _json(request)
    return {
        "ok": True,
        "ignored": True,
        "reason": "Legacy auth provider webhook retired; Keycloak is the only auth provider",
    }


@router.post("/memory-extraction")
@router.post("/memory-extraction/benchmark-locomo")
@router.post("/memory-user-memory/persona/update-writing")
@router.post("/memory-user-memory/pipelines/extract/chat-topic/cancel")
async def retired_memory_webhook(request: Request):
    await _json(request)
    return JSONResponse(
        status_code=status.HTTP_410_GONE,
        content={
            "ok": False,
            "reason": "Legacy TS webhook transport retired; use /api/workflows routes",
        },
    )


@router.post("/video/{provider}")
async def retired_video_callback(provider: str, request: Request):
    await _json(request)
    return JSONResponse(
        status_code=status.HTTP_410_GONE,
        content={
            "ok": False,
            "provider": provider,
            "reason": "Legacy TS video callback transport retired",
        },
    )
